internal/ethapi/api.go

```diff
@@ -1546,7 +1546,7 @@ func (s *PublicTransactionPoolAPI) GetRawTransactionByHash(ctx context.Context,
 func (s *PublicTransactionPoolAPI) GetTransactionReceipt(ctx context.Context, hash common.Hash) (map[string]interface{}, error) {
 	tx, blockHash, blockNumber, index, err := s.b.GetTransaction(ctx, hash)
 	if err != nil {
-		return nil, nil
+		return nil, err
```

Comment:
Wasn't this meant to mean that "we don't know this tx, so it's not an error, simply return nothing"?

Reply:
Makes sense, ok